eng/versioning.targets

```diff
@@ -23,7 +23,7 @@
   </ItemGroup>
 
   <!-- Adds SupportedOSPlatform attribute for Windows Specific libraries -->
-  <ItemGroup Condition="'$(IsWindowsSpecific)' == 'true' and '$(IsTestProject)' != 'true'">
+  <ItemGroup Condition="('$(IsWindowsSpecific)' == 'true' or '$(TargetsWindows)' == 'true') and '$(IsTestProject)' != 'true'">
```

Comment:
This is for adding `[SupportedOSPlatform("windows")]` attribute in `AssemblyInfo.cs` of windows builds so that the analyzer wouldn't want for windows only API call

Reply:
Where is `TargetsWindows` true when `IsWindowsSpecific` not? Could we instead change when `IsWindowsSpecific` is set to capture those scenarios? What would be the tradeoffs between the two approaches?

Reply:
> Where is TargetsWindows true when IsWindowsSpecific not? 

As you know we added `<IsWindowsSpecific>true</IsWindowsSpecific>` for assemblies only supported on Windows,  but `TargetsWindows` can be true for windows only and multi-targeted assemblies, so if the assembly is multitargeted and has `Windows_NT` target then `TargetsWindows` will be true but `<IsWindowsSpecific>true</IsWindowsSpecific>` will be false. 

> Could we instead change when IsWindowsSpecific is set to capture those scenarios? What would be the tradeoffs between the two approaches?

I don't think so, `IsWindowsSpecific` is for annotating assembly as supported windows everywhere refs/assemblies/assembIyInfo wherever needed, but `TargetsWindows` adding the attribute only for windows targeted builds. With that being said we might could use on only `TargetsWindows`, but i am not sure if this value will be set for windows only assemblies

Reply:
> With that being said we might could use on only `TargetsWindows`, but i am not sure if `TargetsWindows` will always be set for windows only assemblies

If we only use `TargetsWindows` then the Supported attribute is not being generated for ref assemblies even for windows only assemblies, keeping as is